### DISCUSSION TITLE
Quasar transition desn't work

### DISCUSSION THREAD

**Discussion by mecjos** [quasarframework/quasar#12863] [General - Components / Directives / etc] [answered]
Hi, I'm trying to give transition effect to a div as follow.. 

```
<q-item-section class="full-height flex flex-center">
      <div v-if="!buttonText" key="icon" transition-show="flip-right" transition-hide="flip-right"> <img :src="'images/feature/' + nav.img" height="60" /></div>
      <div v-else key="text" transition-show="flip-right" transition-hide="flip-right">Button Title</div>
</q-item-section>
```

I expect a result like on transition samples page in quasar doc.. but nothing happens. Where is my mistake?

**Comment by smolinari** [comment, marked as answer]
@mecjos - Those transitions only work on components that offer the `transition` props (show and hide). 

If you want transition on divs, you'll need to use animations. https://quasar.dev/options/animations

Scott
